packages/ecc-client-lit-ga4gh-tes/src/components/create-run/create-run.ts

```diff
@@ -0,0 +1,79 @@
+import { html, LitElement } from "lit";
+import "@elixir-cloud/design";
+
+export class CreateRun extends LitElement {
+  fields = [
+    {
+      key: "name",
+      label: "Name",
+      type: "text",
+      fieldOptions: {
+        required: true,
+      },
+    },
+    {
+      key: "email",
+      label: "Email",
+      type: "email",
+      fieldOptions: {
+        // required: false,
+      },
+    },
+    {
+      key: "address",
+      label: "Address",
+      type: "array",
+      children: [
+        {
+          key: "street",
+          label: "Street",
+          type: "text",
+          fieldOptions: {
+            // required: false,
+          },
+        },
+        {
+          key: "city",
+          label: "City",
+          type: "text",
+          fieldOptions: {
+            required: true,
+          },
+        },
+        {
+          key: "isPrimary",
+          label: "Primary",
+          type: "switch",
+        },
+      ],
+    },
+    {
+      key: "18+",
+      label: "18+",
+      type: "switch",
+      switchOptions: {
+        default: true,
+      },
+    },
+    {
+      key: "id",
+      label: "ID",
+      type: "file",
+      fieldOptions: {
+        required: true,
+      },
+    },
+  ];
+
+  render() {
+    return html`
+      <ecc-utils-design-form
+        .fields=${this.fields}
+        @form-submit=${(e) => {
+          console.log("form - submitted", e.detail);
+        }}
+      >
+      </ecc-utils-design-form>
+    `;
+  }
```

Comment:
Will the inner working of CreateRuns be abstracted and not in the same file since we are passing the fields?

Reply:
Nope this component will be solely responsible for creating the form & can be used is different services like tes, wes etc.

Reply:
got it :+1